### ISSUE TITLE
Proxy causes Req tests to fail

### ISSUE THREAD

**Issue by jano017** [travis-ci/travis-ci#7497]
Relevant build logs follow:
```
Failures:

  httpbin-tests/Network/HTTP/ReqSpec.hs:104: 
  1) Network.HTTP.Req, receiving request headers back, works
       expected: Object (fromList [("headers",Object (fromList [("Accept-Encoding",String "gzip"),("Foo",String "bar"),("Baz",String "quux"),("Host",String "httpbin.org")]))])
        but got: Object (fromList [("headers",Object (fromList [("X-Request-Id",String "12849265-09fb-4a57-98c9-143a9b0ab37d"),("Connect-Time",String "0"),("Accept-Encoding",String "gzip"),("Connection",String "close"),("Foo",String "bar"),("Baz",String "quux"),("Host",String "httpbin.org"),("Total-Route-Time",String "0"),("Via",String "1.1 vegur")]))])

  httpbin-tests/Network/HTTP/ReqSpec.hs:116: 
  2) Network.HTTP.Req, receiving GET data back, works
       expected: Object (fromList [("args",Object (fromList [])),("url",String "https://httpbin.org/get"),("headers",Object (fromList [("Accept-Encoding",String "gzip"),("Host",String "httpbin.org")]))])
        but got: Object (fromList [("args",Object (fromList [])),("url",String "https://httpbin.org/get"),("headers",Object (fromList [("X-Request-Id",String "dbd2a9b5-aef5-4d47-8346-7c045811bf4c"),("Connect-Time",String "0"),("Accept-Encoding",String "gzip"),("Connection",String "close"),("Host",String "httpbin.org"),("Total-Route-Time",String "0"),("Via",String "1.1 vegur")]))])

  httpbin-tests/Network/HTTP/ReqSpec.hs:131: 
  3) Network.HTTP.Req, receiving POST JSON data back, works
       expected: Object (fromList [("args",Object (fromList [])),("json",String "foo"),("data",String "\"foo\""),("url",String "https://httpbin.org/post"),("headers",Object (fromList [("Content-Type",String "application/json; charset=utf-8"),("Accept-Encoding",String "gzip"),("Host",String "httpbin.org"),("Content-Length",String "5")])),("files",Object (fromList [])),("form",Object (fromList []))])
        but got: Object (fromList [("args",Object (fromList [])),("json",String "foo"),("data",String "\"foo\""),("url",String "https://httpbin.org/post"),("headers",Object (fromList [("Content-Type",String "application/json; charset=utf-8"),("X-Request-Id",String "fd1f70ce-b932-4d1a-b7dd-095c36b1516f"),("Connect-Time",String "0"),("Accept-Encoding",String "gzip"),("Connection",String "close"),("Host",String "httpbin.org"),("Content-Length",String "5"),("Total-Route-Time",String "0"),("Via",String "1.1 vegur")])),("files",Object (fromList [])),("form",Object (fromList []))])

  httpbin-tests/Network/HTTP/ReqSpec.hs:154: 
  4) Network.HTTP.Req, receiving POST data back (multipart form data), works
       expected: Object (fromList [("args",Object (fromList [])),("json",Null),("data",String ""),("url",String "https://httpbin.org/post"),("headers",Object (fromList [("Content-Type",String "multipart/form-data; boundary=----WebKitFormBoundaryaGQTQkXrX2ttxhQC"),("Accept-Encoding",String "gzip"),("Host",String "httpbin.org"),("Content-Length",String "242")])),("files",Object (fromList [])),("form",Object (fromList [("foo",String "foo data!"),("bar",String "bar data!")]))])
        but got: Object (fromList [("args",Object (fromList [])),("json",Null),("data",String ""),("url",String "https://httpbin.org/post"),("headers",Object (fromList [("Content-Type",String "multipart/form-data; boundary=----WebKitFormBoundaryaGQTQkXrX2ttxhQC"),("X-Request-Id",String "898d73ef-fc60-4b0b-afea-050fc551044d"),("Connect-Time",String "0"),("Accept-Encoding",String "gzip"),("Connection",String "close"),("Host",String "httpbin.org"),("Content-Length",String "242"),("Total-Route-Time",String "0"),("Via",String "1.1 vegur")])),("files",Object (fromList [])),("form",Object (fromList [("foo",String "foo data!"),("bar",String "bar data!")]))])

  httpbin-tests/Network/HTTP/ReqSpec.hs:180: 
  5) Network.HTTP.Req, receiving PATCHed file back, works
       expected: Object (fromList [("args",Object (fromList [])),("json",Null),("data",String "User-agent: *\nDisallow: /deny\n"),("url",String "https://httpbin.org/patch"),("headers",Object (fromList [("Accept-Encoding",String "gzip"),("Host",String "httpbin.org"),("Content-Length",String "30")])),("files",Object (fromList [])),("form",Object (fromList []))])
        but got: Object (fromList [("args",Object (fromList [])),("json",Null),("data",String "User-agent: *\nDisallow: /deny\n"),("url",String "https://httpbin.org/patch"),("headers",Object (fromList [("X-Request-Id",String "40f1b7b5-3e9a-45d3-a0a1-e79f096912d6"),("Connect-Time",String "0"),("Accept-Encoding",String "gzip"),("Connection",String "close"),("Host",String "httpbin.org"),("Content-Length",String "30"),("Total-Route-Time",String "0"),("Via",String "1.1 vegur")])),("files",Object (fromList [])),("form",Object (fromList []))])

  httpbin-tests/Network/HTTP/ReqSpec.hs:201: 
  6) Network.HTTP.Req, receiving PUT form URL-encoded data back, works
       expected: Object (fromList [("args",Object (fromList [])),("json",Null),("data",String ""),("url",String "https://httpbin.org/put"),("headers",Object (fromList [("Content-Type",String "application/x-www-form-urlencoded"),("Accept-Encoding",String "gzip"),("Host",String "httpbin.org"),("Content-Length",String "18")])),("files",Object (fromList [])),("form",Object (fromList [("foo",String "bar"),("baz",String "5"),("quux",String "")]))])
        but got: Object (fromList [("args",Object (fromList [])),("json",Null),("data",String ""),("url",String "https://httpbin.org/put"),("headers",Object (fromList [("Content-Type",String "application/x-www-form-urlencoded"),("X-Request-Id",String "371cb8ce-37ce-499f-8680-16a7dd8fff00"),("Connect-Time",String "1"),("Accept-Encoding",String "gzip"),("Connection",String "close"),("Host",String "httpbin.org"),("Content-Length",String "18"),("Total-Route-Time",String "0"),("Via",String "1.1 vegur")])),("files",Object (fromList [])),("form",Object (fromList [("foo",String "bar"),("baz",String "5"),("quux",String "")]))])
```
Tl;dr, (from what I suspect), as shown by snippets such as `("Via",String "1.1 vegur")`, there is a proxy on travis-ci builds adding headers to http requests. As the tests from Req check for headers verbatim, the headers added by vegur cause the tests to fail. Also possibly relevant to the Req project, so I'll probably open an issue on that repository.

**Comment by BanzaiMan** [comment]
Hello. What proxy could this be? I do not think we set up any proxy for our builds.

**Comment by jano017** [comment]
Given the "Via: 1.1 vegur" header, I would guess this: https://github.com/heroku/vegur

**Comment by BanzaiMan** [comment]
What I meant was, what is responsible for having a proxy configured between your HTTP client and the HTTP server you are testing?

**Comment by BanzaiMan** [comment]
Closing, due to lack of information.
